src/components/DataLoader.vue

```diff
@@ -0,0 +1,48 @@
+<template>
```

Comment:
This component implements a common pattern that showed up in a handful of pages. It shows a loading indicator if data is loading, an error message if there was a problem loading the data, the `NotFound` component if the data was not found. Otherwise, if the data has been loaded, the page content is displayed in the slot.